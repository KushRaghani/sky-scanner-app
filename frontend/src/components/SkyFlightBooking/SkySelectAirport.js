import React from "react";
import PropTypes from "prop-types";

function SkySelectAirport({
  name,
  onChangeOfAirport,
  optionsData,
  selectedAirport
}) {
  const newOptionsData = [
    { airportName: "Select Airport", airportCode: "" },
    ...optionsData
  ];
  // Render Source and Destination Airports
  return (
    <select
      className="custom-select"
      name={name}
      onChange={onChangeOfAirport}
      value={selectedAirport}
    >
      {newOptionsData.map(data => {
        return (
          <option key={data.airportCode} value={data.airportCode}>
            {data.airportName}
          </option>
        );
      })}
    </select>
  );
}

SkySelectAirport.propTypes = {
  name: PropTypes.string.isRequired,
  onChangeOfAirport: PropTypes.func.isRequired,
  optionsData: PropTypes.arrayOf(
    PropTypes.shape({
      airportCode: PropTypes.string,
      airportName: PropTypes.string
    })
  ).isRequired,
  selectedAirport: PropTypes.string.isRequired
};

export default SkySelectAirport;
