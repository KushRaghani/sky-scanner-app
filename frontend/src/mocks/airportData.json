{
  "info": [
    {
      "airportCode": "KUL-sky",
      "airportName": "Kuala Lumpur - International Airport - KUL",
      "isSource": true,
      "isDestination": true,
      "destinations": ["SIN-sky", "SFO-sky"]
    },
    {
      "airportCode": "SIN-sky",
      "airportName": "Singapore - Changi - SIN",
      "isSource": true,
      "isDestination": true,
      "destinations": ["KUL-sky"]
    },
    {
      "airportCode": "SFO-sky",
      "airportName": "San Francisco - International Airport, SA - SFO",
      "isSource": false,
      "isDestination": true,
      "destinations": []
    }
  ],
  "airports": [
    {
      "name": "Kuala Lumpur - International Airport - KUL",
      "code": "KUL-sky"
    },
    {
      "name": "Singapore - Changi - SIN",
      "code": "SIN-sky"
    },
    {
      "name": "San Francisco - International Airport, SA - SFO",
      "code": "SFO-sky"
    }
  ],
  "flights": [
    {
      "source": "KUL-sky",
      "destination": "SIN-sky"
    },
    {
      "source": "KUL-sky",
      "destination": "SFO-sky"
    },
    {
      "source": "SIN-sky",
      "destination": "KUL-sky"
    }
  ],
  "source": [
    {
      "name": "Kuala Lumpur - International Airport - KUL",
      "code": "KUL-sky"
    },
    {
      "name": "Singapore - Changi - SIN",
      "code": "SIN-sky"
    }
  ],
  "destination": [
    {
      "name": "Kuala Lumpur - International Airport - KUL",
      "code": "KUL-sky"
    },
    {
      "name": "Singapore - Changi - SIN",
      "code": "SIN-sky"
    },
    {
      "name": "San Francisco - International Airport, SA - SFO",
      "code": "SFO-sky"
    }
  ]
}
