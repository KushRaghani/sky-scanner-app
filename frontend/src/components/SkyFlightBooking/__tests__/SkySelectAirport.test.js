import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import airportData from "../../../mocks/airportData";
import SkySelectAirport from "../SkySelectAirport";

function loadSkySelectAirport(args) {
  const defaultProps = {
    optionsData: [],
    name: "",
    onChangeOfAirport: () => {},
    selectedAirport: ""
  };

  const props = { ...defaultProps, ...args };
  return <SkySelectAirport {...props} />;
}
describe("SkySelectAirport", () => {
  // Enyzme tests
  test("renders select on load with no optionsData", () => {
    const wrapper = shallow(loadSkySelectAirport());
    expect(wrapper.find("select").length).toBe(1);
    expect(wrapper.find("option").length).toBe(1);
    expect(wrapper.find("option").text()).toBe("Select Airport");
  });

  test("renders select and option with mock optionsData", () => {
    const wrapper = shallow(
      loadSkySelectAirport({ optionsData: airportData.source })
    );
    expect(wrapper.find("select").length).toBe(1);
    expect(wrapper.find("option").length).toBe(3);
  });

  test("renders select and option with mock optionsData", () => {
    const wrapper = shallow(
      loadSkySelectAirport({ optionsData: airportData.source })
    );
    expect(wrapper.find("select").length).toBe(1);
    expect(wrapper.find("option").length).toBe(3);
  });

  // Snapshot testing
  test("snapshot with default data", () => {
    const tree = renderer.create(loadSkySelectAirport());
    expect(tree).toMatchSnapshot();
  });
});
