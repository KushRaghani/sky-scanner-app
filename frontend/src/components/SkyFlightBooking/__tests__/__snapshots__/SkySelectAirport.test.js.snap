// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkySelectAirport snapshot with default data 1`] = `
<select
  className="custom-select"
  name=""
  onChange={[Function]}
  value=""
>
  <option
    value=""
  >
    Select Airport
  </option>
</select>
`;
